tensorboard/webapp/webapp_data_source/tb_feature_flag_data_source.ts

```diff
@@ -16,9 +16,16 @@ import {Injectable} from '@angular/core';
 
 import {TBFeatureFlagDataSource} from './tb_feature_flag_data_source_types';
 
+/**
+ * Save the initial URL query params, before the AppRoutingEffects initialize.
+ * Ideally, AppRouting would manage all URL params.
+ * https://github.com/tensorflow/tensorboard/issues/4207
```

Comment:
You can drop this line. Even if router gains the capability, we want this component to read from the URL to bootstrap certain feature flags states.

Reply:
Done, thanks!